newsView.test.js

```diff
@@ -0,0 +1,66 @@
+/**
+* @jest-environment jsdom
+*/
+
+const fs = require('fs');
+const { networkInterfaces } = require('os');
+const NewsView = require('./newsView')
+  
+describe ("News View", () => {
+  it("displays headlines in the browser", () => {
+    document.body.innerHTML = fs.readFileSync('./index.html');
+
+    const newsModel = { 
+      getHeadlines: () => ['testing', 'testing 2'],
+      getLinks: () => [],
+      getImages: () => []
+    }
+    const newsView = new NewsView(newsModel);
+  
+    newsView.displayNews();
+  
+    expect(document.querySelectorAll('div.headline').length).toBe(2);
+  });
+
+  it("links to the articles", () => {
+    document.body.innerHTML = fs.readFileSync('./index.html');
+    
+    const newsModel = { 
+      getHeadlines: () => [1],
+      getLinks: () => ['https://www.testing.com/'],
+      getImages: () => []
+    };
+    const newsView = new NewsView(newsModel);
+  
+    newsView.displayNews();
+    
+    const headline = document.querySelector('div.headline');
+    expect(headline.querySelectorAll('a').length).toBe(1);
+  });
+
+  it("displays images", () => {
+    document.body.innerHTML = fs.readFileSync('./index.html');
+    
+    const newsModel = { 
+      getHeadlines: () => [1],
+      getLinks: () => [],
+      getImages: () => ['test.jpg']
+    };
+    const newsView = new NewsView(newsModel);
+  
+    newsView.displayNews();
+    
+    const headline = document.querySelector('div.headline');
+    expect(headline.querySelectorAll('img').length).toBe(1);
+  });
+
+  it("has a search function", () => {
```

Comment:
To test the search function, you can mock the result returned by the `NewsApi` class when searching with a term — by injecting a "mock" into the view (using the second argument):

```js
const mockedStories = [ { webTitle: 'Extra, extra, read all about it! Test coverage exceeding 100%!!', webUrl: 'https://www.testing.com', fields: { thumbnail: 'testing.png' } }];

// mockApi is just an object with a function getNews() that we're
// mocking - it will call the given callback with our mocked stories

const mockApi = {
  getNews: (term, callback) => {
      callback(mockedStories)
  }
}
const newsView = new NewsView(newsModel, mockApi);

// ...

// setting the search input and clicking the button
searchInput.value = 'random search';
searchButton.click()

Reply:
You can also use [Jest automatic class mocks](https://jestjs.io/docs/es6-class-mocks#automatic-mock) (or manual ones), which provide a bit more (you can verify that `NewsApi.getNews()` has been called with the correct search term, mock its behaviour, etc)